src/Microsoft.Performance.SDK.Runtime/DTO/TableConfigurationsSerializer.cs

```diff
@@ -94,15 +96,32 @@ public class TableConfigurationsSerializer
 
                 var desrailizedObject = serializer.ReadObject(stream);
 
-                if (desrailizedObject is ISupportUpgrade<PrebuiltConfigurations> supportUpgrade)
+                if (desrailizedObject is not PrebuiltConfigurationsBase configurationsBase)
                 {
-                    prebuiltConfigurations = supportUpgrade.Upgrade();
+                    return Enumerable.Empty<Processing.TableConfigurations>();
                 }
-                else if (desrailizedObject is PrebuiltConfigurations latestConfigs)
+
+                // Ensure we don't get stuck in a loop by tracking visited configurations
+                HashSet<PrebuiltConfigurationsBase> visited = new HashSet<PrebuiltConfigurationsBase>();
+
+                while (visited.Add(configurationsBase))
                 {
-                    prebuiltConfigurations = latestConfigs;
+                    if (configurationsBase is ISupportUpgrade<V1_0.PrebuiltConfigurations> v1_0)
```

Comment:
Is it necessary to distinguish the actual version it supports? How about 
```C#
interface UpgradableConfiguration {
  ConfigurationBase Upgrade();
}
```

Reply:
Other classes implement `ISupportUpgrade` where `T` is relevant. However, I made an abstract class that will help with this logic and do what you're suggesting.